### PR TITLE
Planck swap hand matrix for rev6, fixing issue #17388

### PR DESCRIPTION
**Pull request by jenspets**
The swap hand matrix gave compile errors for rev6, as the matrix dimensions were changed. This PR corrects the matrix for rev6 boards. See issue #17388

## Description

Added a preprocessor directive for rev6 and rev6_drop boards in the file keyboards/planck/planck.c, which sets the correct swap_hand_config matrix for these revisions. 
 
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #17388

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jenspets** [comment]
> If anything, this should be moved to each revision, as I'm sure the planck ez also needs to be changed.

Do you propose that I move the declaration of the variable from the original file (keyboards/planck/planck.c) to a source file in each revision subdirectory? Or should I just add the missing revision to the preprocessor branch test in keyboards/planck/planck.c?

I could also reverse the logic, testing for the older board revisions instead, such that all new revisions that are based on rev6 will build without adding the revision to the list of revisions in preprocessor statement.

**Comment by jenspets** [comment]
All revisions build without error:
```
$ echo SWAP_HANDS_ENABLE = yes >> keyboards/planck/keymaps/default/rules.mk
$ REVS="rev1 rev2 rev3 rev4 rev5 rev6 rev6_drop light ez thk"
$ for i in $REVS; do make planck/$i:default > /dev/null 2>&1; if [[ $? -eq 0 ]]; then echo -e "\033[01;32m$i: OK\033[00m" ; else echo -e "\033[01;31m$i: ERROR\033[00m" ; fi; done
rev1: OK
rev2: OK
rev3: OK
rev4: OK
rev5: OK
rev6: OK
rev6_drop: OK
light: OK
ez: OK
thk: OK
```

File: keyboards/planck/planck.c [hunk 1]
```diff
@@ -2,12 +2,31 @@
 
 #ifdef SWAP_HANDS_ENABLE
 __attribute__ ((weak))
+#    if defined(KEYBOARD_planck_rev1) || defined(KEYBOARD_planck_rev2) || defined(KEYBOARD_planck_rev3) \
```
[the remark this answers is not in the document]
Comment (jenspets) [reply keyboards/planck/planck.c]: I moved the hand_swap_config matrix to each individual revision in commit [679247b](https://github.com/qmk/qmk_firmware/pull/17389/commits/679247bd680c086859f0206b85fb296c7af1f7ec). It looks cleaner, but new revisions need to declare this in their revision source file for swap_hands to work.

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**